MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/activity/login/LoginActivity.kt

```diff
@@ -0,0 +1,167 @@
+        }
+        MaterialTheme {
+            Box(
+                modifier = Modifier.semantics {
+                    // Allows to use testTag() for UiAutomator's resource-id.
+                    // It can be enabled high in the compose hierarchy,
+                    // so that it's enabled for the whole subtree
+                    testTagsAsResourceId = true
+                }
+            ) {
+
+                Column(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    TopAppBar(title = {
+                        Text(text = "Please login to this sample")
+                    })
+                    InputField(
+                        title = "User: ",
+                        value = userName,
+                        tag = "userName",
+                        onValueChange = { userName = it })
+                    InputField(
+                        title = "Password: ",
+                        value = password,
+                        tag = "password",
+                        onValueChange = {
+                            password = it
+                        },
+                        visualTransformation = PasswordVisualTransformation(),
+                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password)
+                    )
+                    Button(
+                        modifier = Modifier.testTag("login"),
+                        enabled = userName.isNotEmpty() && password.isNotEmpty(),
+                        onClick = {
+                            sampleViewModel.login(userName, password)
+                            startActivity(
+                                Intent(
+                                    this@LoginActivity,
+                                    MainActivity::class.java
+                                )
+                            )
+                            finish()
+                        }) {
+                        Text("Login")
+                    }
+                    Text(
+                        text = "Do not put any actual userdata here, only data for testing purposes.",
+                        modifier = Modifier.padding(4.dp)
+                    )
+                }
+            }
+        }
+        fullyDrawnReporter.removeReporter()
+    }
+
+    @Composable
+    fun InputField(
+        title: String,
+        value: String,
+        tag: String,
+        onValueChange: (String) -> Unit,
+        visualTransformation: VisualTransformation = VisualTransformation.None,
+        keyboardOptions: KeyboardOptions = KeyboardOptions()
+    ) {
+        Surface(shape = MaterialTheme.shapes.small, modifier = Modifier.padding(4.dp)) {
+            Text(text = title, modifier = Modifier.padding(start = 4.dp))
+            TextField(
+                modifier = Modifier.testTag(tag),
```

Comment:
Can we leverage the default `label` + modifier of TextField?
In this case you don't need the `Surface`, nor the `Text` composables

```
TextField(
    label = { Text(title) },
    modifier = Modifier.testTag(tag).padding(4.dp),
```